webhooks/finished/server.js

```diff
@@ -0,0 +1,407 @@
+    console.log(basicLinkTokenObject);
+    res.json(response.data);
+  } catch (error) {
+    console.log(`Running into an error!`);
+    next(error);
+  }
+});
+
+/**
+ * Swap the public token for an access token, so we can access transaction info
+ * in the future
+ */
+app.post("/server/swap_public_token", async (req, res, next) => {
+  try {
+    const response = await plaidClient.itemPublicTokenExchange({
+      public_token: req.body.public_token,
+    });
+    console.log(`You got back ${JSON.stringify(response.data)}`);
+    await updateUserRecord(FIELD_ACCESS_TOKEN, response.data.access_token);
+    await updateUserRecord(FIELD_USER_STATUS, "connected");
+
+    res.json({ status: "success" });
+  } catch (error) {
+    next(error);
+  }
+});
+
+/**
+ * Grabs transaction info for the user and return it as a big ol' JSON object
+ */
+app.get("/server/transactions", async (req, res, next) => {
+  try {
+    const access_token = await userRecord[FIELD_ACCESS_TOKEN];
+    const startDate = moment().subtract(30, "days").format("YYYY-MM-DD");
+    const endDate = moment().format("YYYY-MM-DD");
+
+    const transactionResponse = await plaidClient.transactionsGet({
+      access_token: access_token,
+      start_date: startDate,
+      end_date: endDate,
+      options: { count: 10 },
+    });
+    res.json(transactionResponse.data);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Fetches balance data
+app.get("/server/balances", async (req, res, next) => {
+  try {
+    const access_token = userRecord[FIELD_ACCESS_TOKEN];
+    const balanceResponse = await plaidClient.accountsBalanceGet({
+      access_token: access_token,
+      options: {
+        min_last_updated_datetime: "2020-01-01T00:00:00Z",
+      },
+    });
+    res.json(balanceResponse.data);
+  } catch (error) {
+    next(error);
+  }
+});
+
+/**
+ * Kicks off the request to create an asset report. In non-sandbox mode
+ * this could take several minutes to complete.
+ */
+app.get("/server/create_asset_report", async (req, res, next) => {
+  try {
+    const access_token = userRecord[FIELD_ACCESS_TOKEN];
+    const reportResponse = await plaidClient.assetReportCreate({
+      access_tokens: [access_token],
+      days_requested: 30,
+      options: {
+        user: { first_name: "Jane", last_name: "Foster" },
+        webhook: webhookUrl,
+      },
+    });
+    res.json(reportResponse.data);
+  } catch (error) {
+    next(error);
+  }
+});
+
+/**
+ * Ask Plaid to fire off a new webhook. Useful for testing webhooks... and
+ * not much else.
+ */
+app.post("/server/fire_test_webhook", async (req, res, next) => {
+  try {
+    const access_token = userRecord[FIELD_ACCESS_TOKEN];
+    const webhookResponse = await plaidClient.sandboxItemFireWebhook({
+      access_token: access_token,
+      webhook_type: WebhookType.Item,
+      webhook_code:
+        SandboxItemFireWebhookRequestWebhookCodeEnum.NewAccountsAvailable,
```

Comment:
as i said in my earlier comment, could expand this to not be hardcoded, but built to take different webhooks.

Reply:
Yeah, you can see my thoughts in the earlier comment. Maybe if they expand the fire_webhook endpoint to support more types of webhooks, we can revisit this.